### PR TITLE
seek for prev

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
`seek_for_prev` locate to the position at the last key in the source that at or before target.

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
@siddontang @BusyJay @disksing PTAL

File: test/test_iterator.rs [hunk 1]
```diff
@@ -113,6 +113,32 @@ pub fn test_iterator() {
...
+        iter.seek_for_prev(SeekKey::Key(b"k1-2"));
+        assert!(iter.valid());
+        assert_eq!(iter.key(), b"k1-1");
+        assert_eq!(iter.value(), b"b");
+
+        let mut iter = db.iter();
+        iter.seek_for_prev(SeekKey::Key(b"k1-3"));
+        assert!(iter.valid());
+        assert_eq!(iter.key(), b"k1-3");
+        assert_eq!(iter.value(), b"d");
```
Comment (BusyJay) [inline test/test_iterator.rs]: Missing case for `SeekKey::Start` and `SeekKey::End`.

**Comment by siddontang** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**